[#8] 피드 글 등록 API 컨트롤러 테스트

grey920 opened the pull request:
## #️⃣연관된 이슈
-  #8 

## 📝작업 내용
- PostController 테스트 클래스 생성(c73ccd5a)
- SpringSecurity 적용부분에서 발생한 이슈 해결 (cfb9c4a4)
- 불필요한 JwtSecurityConfig 클래스 제거(f08eddcc)

grey920 committed:
feat(feed): [#8] 글 등록 API 테스트

- pom.xml : spring-security-test 의존성 추가
- JpaConfiguration, MusolmateApplication : @WebMvcTest 사용으로 인한 변경

grey920 committed:
fix(member): [#8] @Authentication에 인증정보가 null로 들어오는 이슈로 인해 Member 객체가 UserDetails를 구현하도록 변경

grey920 committed:
fix(security): [#8] JwtSecurityConfig 제거 및 PostControllerTest 수정

gyuwon commented on src/main/java/com/flab/musolmate/common/JpaConfiguration.java:
```diff
@@ -0,0 +1,8 @@
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
+@EnableJpaAuditing
```
import 문과 애노테이션 사이에 한줄 띄워주면 가동성이 조금 올라갈 것 같습니다.

gyuwon commented on src/test/java/com/flab/musolmate/post/web/PostControllerTest.java:
```diff
@@ -0,0 +1,107 @@
+    @MockBean(name = "userDetailsService")
+    CustomUserDetailService customUserDetailService;
+
+    static final String PATH_POST_BASE = "/posts";
+    static final String PATH_POST_REGISTER = "/register";
```
두 상수를 합쳐서 그냥 `PATH` 라고 해도 클래스의 맥락이 있으니 기독성은 그리 떨어지지 않을 것 같습니다.

gyuwon commented on src/test/java/com/flab/musolmate/post/web/PostControllerTest.java:
```diff
@@ -0,0 +1,107 @@
+            .email( "admin@gmail.com" )
+            .password( "1234qwer" )
+            .nickName( "aaa" )
+//            .authorities( Set.of( authAdmin, authUser ) )
```
이 주석은 어떤 의미인가요?

grey920 replied:
ROLE_USER 권한이 **없는** 사용자일 경우 어떻게 실패하는지 테스트해보려고 기존에 설정해두었던 권한들을 잠시 주석해놓았습니다..!

<img width="1312" alt="image" src="https://github.com/f-lab-edu/music-soulmate/assets/58028215/17e08e10-0c19-460e-8409-470a64e5461d">
<img width="1302" alt="image" src="https://github.com/f-lab-edu/music-soulmate/assets/58028215/f34fc6ef-e610-4115-84c9-339a694706b4">

gyuwon commented on src/test/java/com/flab/musolmate/post/web/PostControllerTest.java:
```diff
@@ -0,0 +1,107 @@
+        ObjectMapper objectMapper = new ObjectMapper();
+        String jsonContents = objectMapper.writeValueAsString( postRegisterRequest );
+
+        // when
```
`// when / then` 이라고 하면 더 좋을 것 같아요.

gyuwon commented:
PR 설명과 코드를 읽었을 때 전반적으로 무엇을 하려는 것인지 잘 와 닿지 않습니다. 사후 테스트 하나 추가하는 것이 목적인가요?

grey920 committed:
refactor(test): [#10] 코멘트 반영

- 가독성 위해 import문 아래 공백 추가
- PATH 상수 하나로 합침
- 권한 테스트한 주석 되살리기
- when / then 주석 수정

grey920 commented:
https://github.com/f-lab-edu/music-soulmate/pull/10#issuecomment-1949886181

추가한 PostController의 엔드포인트가 정상적으로 동작하는지 확인하는 것이 목적입니다!
- 지정한 URL에 접근하는지
- 요청 데이터인 Member와 PostRegisterRequest를 제대로 검증하는지
- 정상적으로 접근했을 때 의도한 응답코드를 반환하는지

이 중 요청 데이터인 Member 객체가 의도한대로 값이 들어오지 않아 관련된 수정 작업이 있었습니다.( cfb9c4a4 )

grey920 commented:
> PR 설명과 코드를 읽었을 때 전반적으로 무엇을 하려는 것인지 잘 와 닿지 않습니다. 사후 테스트 하나 추가하는 것이 목적인가요?

추가한 PostController의 엔드포인트가 정상적으로 동작하는지 확인하는 것이 목적입니다!

- 지정한 URL에 접근하는지
- 요청 데이터인 Member와 PostRegisterRequest를 제대로 검증하는지
- 정상적으로 접근했을 때 의도한 응답코드를 반환하는지

이 중 요청 데이터인 Member 객체가 의도한대로 값이 들어오지 않아 관련된 수정 작업이 있었습니다.( https://github.com/f-lab-edu/music-soulmate/commit/cfb9c4a4eee7b69313c0e02578e7adaac70e8978 )
